### PR TITLE
Fix: proxyType name fix for remove proxy confirmation screen

### PR DESCRIPTION
**Pull request by egor0798**
Split camel case string for unknows proxyType on remove proxy confirmation screen

### CODE REVIEW

**Review by Asmadek — APPROVED** [review]

File: src/renderer/shared/lib/utils/strings.ts [hunk 1]
```diff
@@ -105,3 +105,12 @@ export const toFixedNotation = (value: number, maxPrecision = 20): string => {
 
   return value.toFixed(trailingZeros);
 };
+
+/**
+ * Splits string in camelCase by capital letters and adds spaces
+ * @param value string in camel case
+ * @return {String}
+ */
+export const splitCamelCaseString = (value: string): string => {
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/strings.ts:114]: Please add tests

File: src/renderer/widgets/RemoveProxy/ui/Confirmation.tsx [hunk 2]
```diff
@@ -64,7 +65,9 @@ export const Confirmation = ({
         <hr className="border-filter-border w-full pr-2" />
 
         <DetailRow label={t('proxy.details.revokeAccessType')} className="pr-2">
-          <FootnoteText>{t(ProxyTypeName[proxyAccount.proxyType])}</FootnoteText>
+          <FootnoteText>
+            {t(ProxyTypeName[proxyAccount.proxyType]) || splitCamelCaseString(proxyAccount.proxyType)}
```
Comment (tuul-wq) [inline src/renderer/widgets/RemoveProxy/ui/Confirmation.tsx:69]: I still don't understand why not to use `t(proxyUtils.getProxyTypeName(proxyType))`?
For not existing i18n token, `t()` will return original argument, which is the result of `proxyUtils.getProxyTypeName`

```javascript
function getProxyTypeName(proxyType: ProxyType | string): string {
  return ProxyTypeName[proxyType as ProxyType] || splitCamelCaseString(proxyType as string);
}
```

File: src/renderer/shared/lib/utils/__tests__/strings.test.ts [hunk 3]
```diff
@@ -13,5 +13,9 @@ describe('shared/lib/utils/strings', () => {
     test('split camel case for section and method', () => {
       expect(formatSectionAndMethod('simpleProxy', 'addProxy')).toEqual('Simple proxy: Add proxy');
     });
+
+    test('split camel case sting and add spaces', () => {
```
Comment (tuul-wq) [inline src/renderer/shared/lib/utils/__tests__/strings.test.ts:17]:
```suggestion
    test('split camel case string into parts divided by space', () => {
```

File: src/renderer/entities/proxy/lib/utils.ts [hunk 4]
```diff
@@ -124,9 +122,7 @@ const ProxyTypeName: Record<ProxyType, string> = {
   [ProxyType.NOMINATION_POOLS]: 'proxy.names.nominationPools',
 };
 
-function getProxyTypeName(proxyType: ProxyType | string, t: TFunction) {
+function getProxyTypeName(proxyType: ProxyType | string): string {
   // if proxy type is not in ProxyTypeName enum split camel case string and add spaces
```
Comment (tuul-wq) [inline src/renderer/entities/proxy/lib/utils.ts:126]: This comment redundant because we have a properly called `splitCamelCaseString `

**Review by tuul-wq — APPROVED** [review]

**Pull request merged**